packages/aws-rfdk/lib/lambdas/nodejs/cert-rotation-monitor/index.ts

```diff
@@ -0,0 +1,81 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/* eslint-disable no-console */
+
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { SecretsManager, CloudWatch } from 'aws-sdk';
+
+import {  Secret } from '../lib/secrets-manager';
+import { Certificate } from '../lib/x509-certs';
+
+const SECRETS_MANAGER_VERSION = '2017-10-17';
+
+async function getSecretsByTag(secretsManager: SecretsManager, uniqueID: string): Promise<SecretsManager.SecretListType | undefined> {
+  console.debug(`getSecretsByTag for secrets with tag ${uniqueID}`);
+  // Looking for all certificates that contain tags with uniqueID.
+  const request: SecretsManager.ListSecretsRequest = {
+    Filters: [{
+      Key: 'tag-value',
+      Values: [uniqueID],
+    }],
+  };
+  try {
+    const response: SecretsManager.ListSecretsResponse = await secretsManager.listSecrets(request).promise();
+    return response.SecretList;
+  } catch (e) {
+    throw new Error(`getSecretsByTag '${uniqueID}' failed':` +
+      `${e.code} -- ${e.message}`);
+  }
+}
+
+function putValueToMetric(value: number){
+  const metric = {
+    MetricData: [
+      {
+        MetricName: 'DaysToExpiry',
+        Dimensions: [
+          {
+            Name: 'Certificate Metrics',
+            Value: process.env.UNIQUEID!,
+          },
+        ],
+        Unit: 'None',
+        Value: value,
+      },
+    ],
+    Namespace: 'AWS/RFDK',
+  };
+  const cloudwatch = new CloudWatch();
+  cloudwatch.putMetricData(metric, (err) => {
+    if (err) {
+      console.log(`Failed to put value to CertDaysToExpiry metric: ${err}`);
+    } else {
+      console.log('Value was added to CertDaysToExpiry metric.');
+    }
+  });
+}
+
+/**
+ * Lambda for adding metric about Secret expiration date.
+ *
+ */
+export async function handler(): Promise<void> {
+  const secretsManager = new SecretsManager({ apiVersion: SECRETS_MANAGER_VERSION });
+  const secretsList = await getSecretsByTag(secretsManager, process.env.UNIQUEID!);
+  if (secretsList) {
+    for (const secret of secretsList) {
```

Comment:
Do we ever expect multiple certificates to be attached to a single tag? Looking at the code, it didn't look possible to me. If that's the case, maybe we should check if this list contains <= 1 items.

Thinking it through, if we switch the statistic from `p99` to `minimum` like I suggested in a previous comment, maybe we can leave this to iterate through the items. I think leaving it at `p99` will be a problem though, because if for some reason it did emit 2 metrics for 2 expiry dates, I think p99 would end up looking at the higher value.

Reply:
It is possible if will Lambda generate more than one certificate but we do not expect it so we do not want to have any exceptions in this case. This code will work if we receive one ore more secrets with certificates to check.

Reply:
If we do leave the logic like this where we are looping over the list, but don't expect it to have more than 1 item, maybe we should have a comment saying that we don't expect multiple items, but the monitor should still work if there is more than one since we're alarming on the minimum value.